src/optimum/nvidia/models/whisper.py

```diff
@@ -0,0 +1,669 @@
+    def supports_strong_typing() -> bool:
+        return False
+
+
+class WhisperDecoderConfig(TensorRTConfig):
+    @classmethod
+    def from_config(cls, config: TransformersPretrainedConfig) -> "TensorRTConfig":
+        # Retrieve the quantization from the transformers config (if provided)
+        _, qconfig = TensorRTConfig.get_quantization_config(config)
+
+        trt_config = cls(
+            architecture=config.architectures[0],
+            dtype=dtype_to_str(config.torch_dtype),  # TODO: always float32?
+            logits_dtype="float32",
+            vocab_size=config.vocab_size,
+            max_position_embeddings=config.max_target_positions,
+            hidden_size=config.d_model,
+            num_hidden_layers=config.decoder_layers,
+            num_attention_heads=config.decoder_attention_heads,
+            num_key_value_heads=config.decoder_attention_heads,
+            hidden_act=config.activation_function,
+            intermediate_size=None,
+            norm_epsilon=None,
+            position_embedding_type="learned_absolute",
+            world_size=1,
+            tp_size=1,
+            pp_size=1,
+            quantization=qconfig,
+            use_parallel_embedding=None,
+            embedding_sharding_dim=None,
+            share_embedding_table=None,
+            head_size=-1,  # We need to set it otherwise TRT-LLM tries to compute `hidden_size // num_attention_heads`
+            max_source_positions=config.max_source_positions,
+            decoder_ffn_dim=config.decoder_ffn_dim,
+            trt_model_class="TrtWhisperDecoderPretrainedModel",
+            trt_model_file=pathlib.Path(__file__),
+            num_encoder_attention_heads=config.encoder_attention_heads,
+        )
+
+        trt_config.mapping.gpus_per_node = min(trt_config.mapping.world_size, 8)
+
+        return trt_config
+
+    def get_plugins_config(self) -> PluginConfig:
+        config = super().get_plugins_config()
+        config.bert_attention_plugin = "disable"
+        config.gpt_attention_plugin = self.dtype
+        config.paged_kv_cache = "disable"  # TODO: getting AssertionError: Paged kv cache is enabled, the kv_cache_block_pointers tensor shall not be None
+
+        config.moe_plugin = "disable"
+        config.gemm_plugin = self.dtype
+
+        return config
+
+    @staticmethod
+    def supports_strong_typing() -> bool:
+        return False
+
+
+class TrtWhisperEncoderPretrainedModel(PretrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.model = TrtWhisperEncoder(
+            n_mels=config.num_mel_bins,
+            n_ctx=config.max_source_positions,
+            n_state=config.hidden_size,
+            n_head=config.num_attention_heads,
+            n_layer=config.num_hidden_layers,
+            dtype=str_dtype_to_trt(config.dtype),
+        )
+
+    def forward(self, **kwargs):
+        return self.model(**kwargs)
+
+    def prepare_inputs(self, max_batch_size=16, **kwargs):
```

Comment:
`PretrainedModel`'s [`prepare_inputs`](https://github.com/NVIDIA/TensorRT-LLM/blob/66ca3378c61efa3154ed34a48cfc362351405eef/tensorrt_llm/models/modeling_utils.py#L407) is no good for whisper. We are forced to comply to its signature though, as this is called [with uncustomizable inputs in TRT-LLM builder.py](https://github.com/NVIDIA/TensorRT-LLM/blob/66ca3378c61efa3154ed34a48cfc362351405eef/tensorrt_llm/builder.py#L643-L657)... not very generalizable